Mathlib/Algebra/Module/LinearMap.lean

```diff
@@ -568,6 +568,21 @@ theorem id_comp : id.comp f = f :=
   LinearMap.ext fun _ ↦ rfl
 #align linear_map.id_comp LinearMap.id_comp
 
+theorem comp_assoc
+    {R : Type*} {R₂ : Type*} {R₃ : Type*} {R₄ : Type*}
+    {M : Type*} {M₂ : Type*} {M₃ : Type*} {M₄ : Type*}
+    [Semiring R] [Semiring R₂] [Semiring R₃] [Semiring R₄]
+    [AddCommMonoid M] [AddCommMonoid M₂] [AddCommMonoid M₃] [AddCommMonoid M₄]
+    [Module R M] [Module R₂ M₂] [Module R₃ M₃] [Module R₄ M₄]
+    {σ₁₂ : R →+* R₂} {σ₂₃ : R₂ →+* R₃} {σ₃₄ : R₃ →+* R₄}
+    {σ₁₃ : R →+* R₃} {σ₂₄ : R₂ →+* R₄} {σ₁₄ : R →+* R₄}
+    [RingHomCompTriple σ₁₂ σ₂₃ σ₁₃] [RingHomCompTriple σ₂₃ σ₃₄ σ₂₄]
+    [RingHomCompTriple σ₁₃ σ₃₄ σ₁₄] [RingHomCompTriple σ₁₂ σ₂₄ σ₁₄]
+    (f : M →ₛₗ[σ₁₂] M₂) (g : M₂ →ₛₗ[σ₂₃] M₃) (h : M₃ →ₛₗ[σ₃₄] M₄) :
+    ((h.comp g : M₂ →ₛₗ[σ₂₄] M₄).comp f : M →ₛₗ[σ₁₄] M₄) = h.comp (g.comp f : M →ₛₗ[σ₁₃] M₃) :=
+  rfl
```

Comment:
Can this statement be golfed by reusing a bunch of variables?

Reply:
Probably yes, but then the order of implicit / typeclass arguments comes out a little strange. I thought it would be safest to keep the exact order that it had before, but it probably doesn't matter enough if you prefer otherwise.

Reply:
New signature for reference:
```
LinearMap.comp_assoc.{u_18, u_17, u_12, u_11, u_10, u_9, u_4, u_3, u_2, u_1}
  {R : Type u_1} {R₁ : Type u_2} {R₂ : Type u_3} {R₃ : Type u_4}
  {M : Type u_9} {M₁ : Type u_10} {M₂ : Type u_11} {M₃ : Type u_12}
  [Semiring R] [Semiring R₁] [Semiring R₂] [Semiring R₃]
  [AddCommMonoid M] [AddCommMonoid M₁] [AddCommMonoid M₂] [AddCommMonoid M₃]
  {module_M₁ : Module R₁ M₁} {module_M₂ : Module R₂ M₂} {module_M₃ : Module R₃ M₃}
  {σ₁₂ : R₁ →+* R₂} {σ₂₃ : R₂ →+* R₃} {σ₁₃ : R₁ →+* R₃} [RingHomCompTriple σ₁₂ σ₂₃ σ₁₃]
  {R₄ : Type u_17} {M₄ : Type u_18} [Semiring R₄] [AddCommMonoid M₄] [Module R₄ M₄]
  {σ₃₄ : R₃ →+* R₄} {σ₂₄ : R₂ →+* R₄} {σ₁₄ : R₁ →+* R₄}
  [RingHomCompTriple σ₂₃ σ₃₄ σ₂₄] [RingHomCompTriple σ₁₃ σ₃₄ σ₁₄] [RingHomCompTriple σ₁₂ σ₂₄ σ₁₄]
  (f : M₁ →ₛₗ[σ₁₂] M₂) (g : M₂ →ₛₗ[σ₂₃] M₃) (h : M₃ →ₛₗ[σ₃₄] M₄) :
  comp (comp h g) f = comp h (comp g f)
```

(I'm not sure why the `Module` aren't type class arguments.)